go-controller/pkg/ovn/egressip.go

```diff
@@ -831,11 +841,18 @@ func (oc *Controller) executeCloudPrivateIPConfigOps(egressIPName string, ops ma
 				oc.recorder.Eventf(&eIPRef, kapi.EventTypeWarning, "CloudAssignmentFailed", "egress IP: %s for object EgressIP: %s could not be created, err: %v", egressIP, egressIPName, err)
 				return fmt.Errorf("cloud add request failed for CloudPrivateIPConfig: %s, err: %v", cloudPrivateIPConfigName, err)
 			}
-			// toDelete is non-empty, this indicates an DELETE for which
-			// the object **must** exist, if not: that's an error.
+			// toDelete is non-empty, this indicates a DELETE - if the object does not exist, log an Info message and continue with the next op.
+			// The reason for why we are not throwing an error here is that desired state (deleted) == isState (object not found).
+			// If for whatever reason we have a pending toDelete op for a deleted object, then this op should simply be silently ignored.
+			// Any other error, return an error to trigger a retry.
 		} else if op.toDelete != "" {
 			if err != nil {
-				return fmt.Errorf("cloud deletion request failed for CloudPrivateIPConfig: %s, could not get item, err: %v", cloudPrivateIPConfigName, err)
+				if apierrors.IsNotFound(err) {
+					klog.Infof("Cloud deletion request failed for CloudPrivateIPConfig: %s, item already deleted, err: %v", cloudPrivateIPConfigName, err)
```

Comment:
nit: start log in lower case as with other logs? Your choice but I know you can't un-see now :P

Reply:
You can convince me for errors, but for Info messages we do uppercase  ;-) 
~~~
grep klog.Info -R
(...)
go-controller/pkg/ovn/egressgw.go:	klog.Infof("OVN ECMP route cache is: %+v", ovnRouteCache)
go-controller/pkg/ovn/egressgw.go:	klog.Infof("Cluster ECMP route cache is: %+v", clusterRouteCache)
go-controller/pkg/ovn/egressgw.go:				klog.Infof("Found stale exgw ecmp route, podIP: %s, nexthop: %s, router: %s",
go-controller/pkg/ovn/gateway_cleanup.go:	klog.Infof("Removing DGP %v", nodes)
go-controller/pkg/ovn/gress_policy.go:	klog.Infof("Deleting service %s, possible VIPs: %v from gressPolicy's %s Address Set", service.Name, ips, gp.policyName)
go-controller/pkg/ovn/egressfirewall.go:	klog.Infof("Adding egressFirewall %s in namespace %s", egressFirewall.Name, egressFirewall.Namespace)
go-controller/pkg/ovn/egressfirewall.go:	klog.Infof("Deleting egress Firewall %s in namespace %s", egressFirewallObj.Name, egressFirewallObj.Namespace)
go-controller/pkg/ovn/egressip.go:					klog.Infof("CloudPrivateIPConfig: %s already assigned to node: %s", cloudPrivateIPConfigName, cloudPrivateIPConfig.Spec.Node)
go-controller/pkg/ovn/egressip.go:					klog.Infof("Cloud deletion request failed for CloudPrivateIPConfig: %s, item already deleted, err: %v", cloudPrivateIPConfigName, err)
go-controller/pkg/ovn/egressip.go:			klog.Infof("syncStaleEgressReroutePolicy will delete %s due to no nexthop or stale logical ip: %v", egressIPName, item)
go-controller/pkg/ovn/egressip.go:	klog.Infof("syncStaleEgressReroutePolicy will remove stale nexthops: %+v", logicalRouterPolicyStaleNexthops)
go-controller/pkg/ovn/egressip.go:			klog.Infof("syncStaleSNATRules will delete %s due to logical ip: %v", egressIPName, item)
go-controller/pkg/ovn/egressip.go:			klog.Infof("syncStaleSNATRules will delete %s due to external ip: %v", egressIPName, item)
go-controller/pkg/ovn/egressip.go:	klog.Infof("Patching status on EgressIP %s: %v", name, statusItems)
go-controller/pkg/ovn/egressip.go:				klog.Infof("Successful assignment of egress IP: %s on node: %+v", egressIP, eNode)
go-controller/pkg/ovn/egressip.go:				klog.Infof("Node: %s is detected as reachable and ready again, adding it to egress assignment", nodeName)
go-controller/pkg/ovn/namespace.go:	klog.Infof("[%s] adding namespace", ns.Name)
go-controller/pkg/ovn/namespace.go:		klog.Infof("[%s] adding namespace took %v", ns.Name, time.Since(start))
go-controller/pkg/ovn/namespace.go:			klog.Infof("Namespace %s: ACL logging is set to deny=%s allow=%s", ns.Name, nsInfo.aclLogging.Deny, nsInfo.aclLogging.Allow)
go-controller/pkg/ovn/namespace.go:	klog.Infof("[%s] updating namespace", old.Name)
go-controller/pkg/ovn/namespace.go:				klog.Infof("Namespace %s: NetworkPolicy ACL logging setting updated to deny=%s allow=%s",
go-controller/pkg/ovn/namespace.go:			klog.Infof("Namespace %s: EgressFirewall ACL logging setting updated to deny=%s allow=%s",
go-controller/pkg/ovn/namespace.go:	klog.Infof("[%s] deleting namespace", ns.Name)
go-controller/pkg/ovn/master.go:				klog.Infof("Won leader election; in active mode")
go-controller/pkg/ovn/master.go:				klog.Infof("No longer leader; exiting")
go-controller/pkg/ovn/master.go:					klog.Infof("Lost the election to %s; in standby mode", newLeaderName)
go-controller/pkg/ovn/master.go:		klog.Infof("Stopped leader election")
~~~